apps/admin_api/test/admin_api/v1/controllers/admin_auth/role_controller_test.exs

```diff
@@ -0,0 +1,216 @@
+defmodule AdminAPI.V1.AdminAuth.RoleControllerTest do
+  use AdminAPI.ConnCase, async: true
+  alias EWalletDB.{Membership, Role}
+
+  describe "/role.all" do
+    test "returns a list of roles and pagination data" do
+      response = admin_user_request("/role.all")
+
+      # Asserts return data
+      assert response["success"]
+      assert response["data"]["object"] == "list"
+      assert is_list(response["data"]["data"])
+
+      # Asserts pagination data
+      pagination = response["data"]["pagination"]
+      assert is_integer(pagination["per_page"])
+      assert is_integer(pagination["current_page"])
+      assert is_boolean(pagination["is_last_page"])
+      assert is_boolean(pagination["is_first_page"])
+    end
+
+    test "returns a list of roles according to search_term, sort_by and sort_direction" do
+      insert(:role, %{name: "matched_2"})
+      insert(:role, %{name: "matched_3"})
+      insert(:role, %{name: "matched_1"})
+      insert(:role, %{name: "missed_1"})
+
+      attrs = %{
+        # Search is case-insensitive
+        "search_term" => "MaTcHed",
+        "sort_by" => "name",
+        "sort_dir" => "desc"
+      }
+
+      response = admin_user_request("/role.all", attrs)
+      roles = response["data"]["data"]
+
+      assert response["success"]
+      assert Enum.count(roles) == 3
+      assert Enum.at(roles, 0)["name"] == "matched_3"
+      assert Enum.at(roles, 1)["name"] == "matched_2"
+      assert Enum.at(roles, 2)["name"] == "matched_1"
+    end
```

Comment:
We should start adding a test that uses the match parsers for all the `.all` endpoints.

Reply:
Updated. Please check.